contracts/rewards/src/contract/execute.rs

```diff
@@ -156,20 +179,221 @@ mod test {
         }
     }
 
-    fn create_contract(stored_params: StoredParams) -> Contract<state::MockStore> {
+    /// Test that rewards parameters are updated correctly. In this test we don't change the epoch duration, so
+    /// that computation of the current epoch is unaffected.
+    #[test]
+    fn update_params() {
+        let initial_epoch_num = 1u64;
+        let initial_epoch_start = 250u64;
+        let initial_rewards_per_epoch = 100u128;
+        let initial_participation_threshold = (1, 2);
+        let epoch_duration = 100u64;
+        let mut contract = setup_with_params(
+            initial_epoch_num,
+            initial_epoch_start,
+            epoch_duration,
+            initial_rewards_per_epoch,
+            initial_participation_threshold,
+        );
+
+        // simulate the below tests running at this block height
+        let cur_height = initial_epoch_start + epoch_duration * 10 + 2;
+
+        let new_params = RewardsParams {
+            rewards_per_epoch: cosmwasm_std::Uint256::from(initial_rewards_per_epoch + 100)
+                .try_into()
+                .unwrap(),
+            participation_threshold: (Uint64::new(2), Uint64::new(3)).try_into().unwrap(),
+            epoch_duration: epoch_duration.try_into().unwrap(), // keep this the same to not affect epoch computation
+        };
+
+        // the epoch shouldn't change when the params are updated, since we are not changing the epoch duration
+        let expected_epoch = contract.get_current_epoch(cur_height).unwrap();
```

Comment:
a few things I realized:
- get_current_epoch should return an error if the cur_height is in the past (otherwise the returned value is nonsensical)
- get_current_epoch is a private function and should remain that, it's kind of weird signature doesn't work well for a public interface of the contract
- in general the tests in the rewards module are tied very close to the actual implementation instead of using only public interfaces. I think we'll need to rework them when all contract interfaces are implemented

Reply:
added the error. I think it will be easier to write tests using only public interfaces once all of the methods are implemented (and probably some queries too).